nats-base-client/jsclient.ts

```diff
@@ -182,7 +183,10 @@ export class JetStreamClientImpl extends BaseApiClient
         break;
       } catch (err) {
         const ne = err as NatsError;
-        if (ne.code === "503" && i + 1 < retries) {
+        if (
+          (ne.code === "503" || ne.code === ErrorCode.Timeout) &&
```

Comment:
if JS or the client disconnects, you may get a timeout.

Reply:
Not sure if this is right though, because no-responders tells you that the message has not been received/processed, while a timeout, you don't know for sure. So it is ok to fail to send and prevent a duplicate than risking a duplicate.

Reply:
even if it managed to create the consumer, I wouldn't know what it is, so I will recreate

Reply:
But this is low-level JS publish call, no? So say I call publish() for a message and you get a timeout, you will try to resend that message, and my point is that this is not correct (since user may not have set headers to avoid duplicates)

Reply:
right! - but that is a documentation issue, if they don't set headers to prevent duplicates, there's no restriction in re-inserting.

Reply:
You should simply not try to resend unless it is 503, that's it. At the higher level if you try to add consumer and that fails, that is the thing you need to try again, not at this low-level.

Reply:
But again! there is no need for the library to try to resend. this is not a doc issue, this is library doing something that it should not do... unless I am really missing something here.

Reply:
I think you are right duplicates or extraneous headers would be inserted not expected in the stream